### PR TITLE
Draft for proposal.md

### PR DESCRIPTION
**Pull request by raizo07**

### CODE REVIEW

File: docs/src/proposals.md [hunk 1]
```diff
@@ -1 +1,120 @@
...
+
+## Customization
+
+While the Proposals component is robust in its default form, you may want to customize it to better fit your protocol’s needs. Here are some ways you can customize it:
+
+Custom Voting Logic: This modifies the vote function to implement custom voting logic or add additional checks.
+
+Proposal Validation: This extends the submit_proposal function to include additional validation or metadata for proposals.
+
+Extended Events: Add or modify events to capture more details about proposals and votes.
```
Comment (tensojka) [inline docs/src/proposals.md:109]: The component should not be altered – it should only be mixed into a contract. Only customizations possible currently are to integrate it with Upgrades for custom proposals and in the future, to alter the voting – though this is planned. Perhaps mention that now is a good time to reach out to us regarding any things users would like to see customized

File: docs/src/proposals.md [hunk 2]
```diff
@@ -1 +1,120 @@
...
+
+### Dependencies
+
+The Proposals component does not depend on other Konoha components, making it flexible and easy to integrate.
+
+It interacts with a governance token contract, so ensure your protocol has a compatible ERC-20 governance token.
```
Comment (tensojka) [inline docs/src/proposals.md:115]: the governance token should be a non-transferable one

File: docs/src/proposals.md [hunk 3]
```diff
@@ -1 +1,120 @@
...
+
+The Proposals component offers the following functionalities:
+
+Submit Proposal: This allows the user to submit a new proposal with the provided payload and upgrade type.
+Vote on Proposal: Allows users to vote on existing proposals.
+Get Proposal Details: This allows for the Retrieval of the details of a specific proposal based on it's ID.
+Get Vote Counts: Retrieves the vote counts (yay and nay) for a specific proposal.
+Get Proposal Status: This checks the current status of a proposal (active, passed, or failed).
+Get Live Proposals: This returns all active proposals in an array of IDs.
+Check User Vote: Checks if a specific user has voted on a specific proposal and their voting status.
```
Comment (tensojka) [inline docs/src/proposals.md:20]: This could be a Markdown list

**Review by tensojka — COMMENTED** [review]
Looks good so far!

Please use code formatting code blocks

File: docs/src/proposals.md [hunk 4]
```diff
@@ -11,13 +11,31 @@ It includes functions for voting on proposals, submitting new proposals, and man
...
-- Vote on Proposal: Allows users to vote on existing proposals.
-- Get Proposal Details: This allows for the Retrieval of the details of a specific proposal based on it's ID.
-- Get Vote Counts: Retrieves the vote counts (yay and nay) for a specific proposal.
-- Get Proposal Status: This checks the current status of a proposal (active, passed, or failed).
-- Get Live Proposals: This returns all active proposals in an array of IDs.
-- Check User Vote: Checks if a specific user has voted on a specific proposal and their voting status.
+- `Submit Proposal`: This allows the user to submit a new proposal with the provided payload and upgrade type.
+- `Vote on Proposal`: The Vote on Proposal component allows users to vote on existing proposals.
+- `Get Proposal Details`: This allows for the Retrieval of the details of a specific proposal based on it's ID.
+- `Get Vote Counts`: The `Get Vote Counts` component is used to Retrieve the vote counts (yay and nay) for a specific proposal.
```
Comment (tensojka) [inline docs/src/proposals.md:17]: This is not a component.

**Comment by raizo07** [comment]
> This only explains how to submit a custom proposal, but not how to add one, what types there are, etc. Closing. You also removed the functionality information?

@ Then you'll ask me to add them instead of closing the PR after all the work I have done. That is a bad look. What then happen to all the work and time I've spent on it?

**Comment by tensojka** [comment]
What you wrote will not be merged or used in documentation as it's faster to rewrite it from scratch to get it to an acceptable state. That you will not be able to get it to an acceptable state is a risk you took on when you requested to be assigned to the task.

We have high standards for quality of code and documentation (especially user-facing as that's often the only thing the decisionmakers will see) and these have not been met.

**Comment by raizo07** [comment]
> What you wrote will not be merged or used in documentation as it's faster to rewrite it from scratch to get it to an acceptable state. That you will not be able to get it to an acceptable state is a risk you took on when you requested to be assigned to the task.
> 
> We have high standards for quality of code and documentation (especially user-facing as that's often the only thing the decisionmakers will see) and these have not been met.

@tensojka  I'll be sending a PR for an alternate version I wrote previously. Kindly review, if it doesn't fit the standards, then I'll hands off it.

**Pull request closed without merging**